planner/core/testdata/enforce_mpp_suite_out.json

```diff
@@ -1203,5 +1203,201 @@
+          "          └─HashAgg_30 1.00 mpp[tiflash]  funcs:count(distinct Column#21)->Column#20, funcs:count(distinct Column#23)->Column#22, funcs:sum(Column#17)->Column#24",
+          "            └─Projection_32 16000.00 mpp[tiflash]  Column#17, test.t.a, test.t.b, Column#18, case(eq(Column#18, 1), test.t.a, <nil>)->Column#21, case(eq(Column#18, 2), test.t.b, <nil>)->Column#23",
+          "              └─ExchangeReceiver_34 16000.00 mpp[tiflash]  ",
+          "                └─ExchangeSender_33 16000.00 mpp[tiflash]  ExchangeType: HashPartition, Hash Cols: [name: test.t.a, collate: binary], [name: test.t.b, collate: binary], [name: Column#18, collate: binary]",
+          "                  └─HashAgg_26 16000.00 mpp[tiflash]  group by:Column#26, Column#27, Column#28, funcs:sum(Column#25)->Column#17",
+          "                    └─Projection_39 20000.00 mpp[tiflash]  cast(Column#19, decimal(10,0) BINARY)->Column#25, test.t.a, test.t.b, Column#18",
+          "                      └─Projection_31 20000.00 mpp[tiflash]  test.t.a, test.t.b, test.t.c, Column#18, case(eq(Column#18, 1), test.t.c, <nil>)->Column#19",
+          "                        └─Expand_29 20000.00 mpp[tiflash]  group set num:2, groupingID:Column#18, [{<test.t.a>},{<test.t.b>}]",
+          "                          └─TableFullScan_11 10000.00 mpp[tiflash] table:t keep order:false, stats:pseudo"
+        ],
+        "Warn": null
+      },
+      {
+        "SQL": "select count(distinct a), count(distinct b), sum(c) from t",
+        "Plan": [
+          "5 5 30000"
+        ],
+        "Warn": null
+      },
+      {
+        "SQL": "EXPLAIN select count(distinct a, b), count(distinct b), count(c), sum(d) from t",
+        "Plan": [
+          "TableReader_26 1.00 root  data:ExchangeSender_25",
+          "└─ExchangeSender_25 1.00 mpp[tiflash]  ExchangeType: PassThrough",
+          "  └─Projection_21 1.00 mpp[tiflash]  Column#6, Column#7, Column#8, Column#9",
+          "    └─HashAgg_22 1.00 mpp[tiflash]  funcs:count(distinct test.t.a, test.t.b)->Column#6, funcs:count(distinct test.t.b)->Column#7, funcs:sum(Column#12)->Column#8, funcs:sum(Column#13)->Column#9",
+          "      └─ExchangeReceiver_24 1.00 mpp[tiflash]  ",
+          "        └─ExchangeSender_23 1.00 mpp[tiflash]  ExchangeType: PassThrough",
+          "          └─HashAgg_20 1.00 mpp[tiflash]  group by:Column#16, Column#17, funcs:count(Column#14)->Column#12, funcs:sum(Column#15)->Column#13",
+          "            └─Projection_27 10000.00 mpp[tiflash]  test.t.c, cast(test.t.d, decimal(10,0) BINARY)->Column#15, test.t.a, test.t.b",
+          "              └─TableFullScan_11 10000.00 mpp[tiflash] table:t keep order:false, stats:pseudo"
+        ],
+        "Warn": [
+          "Some grouping sets should be merged",
+          "Some grouping sets should be merged"
+        ]
+      },
+      {
+        "SQL": "select count(distinct a, b), count(distinct b), count(c), sum(d) from t",
```

Comment:
We don't support it at the moment, is it because TiFlash doesn't support?

Reply:
it needs some changes: Expand op produces non-fixed schema length, I will do it next step. Otherwise, this tiFlash pr will become huge.

Reply:
Understood.